### DISCUSSION TITLE
Rubocop CLI with subcommands

### DISCUSSION THREAD

**Discussion by DannyBen** [rubocop/rubocop#11187] [Ideas]
I know this is highly unlikely to happen, but I would like to raise it anyways.

Wouldn't it be much nicer if rubocop CLI have all its options organized into subcommands, instead of having everything done through `--sometimes-awkward-flags`?

I see docker and git as references. This is much more scalable, and makes the interface more like speaking in plain English with the terminal. Less typing, more intuitive, more scalable, more maintainable.

Example commands:

```bash
$ rubocop check
$ rubocop check --format markdown
$ rubocop check Layout/SpaceBeforeSemicolon

$ rubocop fix
$ rubocop fix --except Layout/SpaceBeforeSemicolon
$ rubocop fix Layout/SpaceBeforeSemicolon

$ rubocop server start
$ rubocop server stop
$ rubocop server status

$ rubocop todo

$ rubocop ls

$ rubocop show
$ rubocop show Layout/SpaceBeforeSemicolon

$ rubocop docs
$ rubocop docs Layout/SpaceBeforeSemicolon

$ rubocop init
$ rubocop init --full
```

**Comment by noraj** [comment]
That would probably be easier for exclusive options.
